File: panel/api/app/routers/history.py
from fastapi import APIRouter, Depends, Query

from .. import auth
from ..db.homer import recent_calls

router = APIRouter(prefix="/api/history", tags=["history"], dependencies=[Depends(auth.current_admin)])


@router.get("")
def list_history(limit: int = Query(50, ge=1, le=500)):
    try:
        rows = recent_calls(limit)
    except Exception:
        return []
    for r in rows:
        r["started_at"] = r["started_at"].isoformat() if r.get("started_at") else None
        r["last_seen"] = r["last_seen"].isoformat() if r.get("last_seen") else None
        r["homer_url"] = f"http://localhost:9080/search/result/data?callid={r['callid']}"
    return rows

File: panel/web/src/api/history.ts
import { api } from "./client";

export type HistoryRow = {
  callid: string;
  started_at: string | null;
  last_seen: string | null;
  message_count: number;
  from_user: string | null;
  to_user: string | null;
  homer_url: string;
};

export const list = (limit = 50) => api<HistoryRow[]>(`/api/history?limit=${limit}`);
